src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -323,7 +325,8 @@ private static <T extends Comparable<T>> RangeIterator<T> buildIterator(Operatio
     {
         var builder = op == Operation.OperationType.OR
                       ? RangeUnionIterator.<T>builder(subIterators.size())
-                      : RangeIntersectionIterator.<T>builder(subIterators.size(), Integer.MAX_VALUE);
+                      // By default, pick 2 most selective indexes
```

Comment:
I remembered @mike-tr-adamson tested SAI intersection performance: it got very bad when there were more than 2 indexes

Reply:
My understanding is that if you don't do intersection w/ the iterators here then it gets done when we fetch the rows later.  Which means we'd be giving a too-large resultset to hnsw, so we'd be giving back incorrect results from the limit it performs.   Is that incorrect?

Reply:
pushed https://github.com/datastax/cassandra/pull/624/commits/56cfece56beefcb3d0040ba3b172dc9c49186044 to use all available indexes for ANN and use at most 2 indexes if there is no ANN

Reply:
ship it!